techlibs/quicklogic/synth_quicklogic.cc

```diff
@@ -261,7 +261,9 @@ struct SynthQuickLogicPass : public ScriptPass {
                 run("iopadmap -bits -outpad outpad A:P -inpad inpad Q:P -tinoutpad bipad EN:Q:A:P A:top");
             } else {
                 run("clkbufmap -buf $_BUF_ Y:A -inpad ck_buff Q:A");
-                run("iopadmap -bits -outpad out_buff A:Q -inpad in_buff Q:A -toutpad EN:A:Q A:top");
+                run("iopadmap -bits -outpad $__out_buff A:Q -inpad $__in_buff Q:A -toutpad EN:A:Q A:top");
```

Comment:
In the `iopadmap` command you are missing a cell type for `-toutpad`. As far as i know AP2 does not have 3-state outputs. In that case I'd suggest removing `-toutpad EN:A:Q` completely. Also `tribuf -logic` in the "prepare" section should be removed for AP2 as well.

Reply:
Made changes